### ISSUE TITLE
mislabeling of "BH" remnants as "NS"

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON-public#413]
I thought this issue had previously been fixed, but I am finding that SN remnants that should definitely be BHs (mass >> 2.5 Msun) are being labeled as NSs, and then this label is changed later-on in the history:

<img width="767" alt="Screen Shot 2024-10-15 at 11 14 50 AM" src="https://github.com/user-attachments/assets/61b1a3d6-e35a-482f-9e4d-8a43f466a5fd">

Here are the initial conditions for re-running this binary with `/projects/b1119/cliotine/pulsar_dev/pulsar_pops/runs/oct_2024/Sukhbold+16/grid_reruns/params.ini`:

```
sim_kwargs_IF = simprop_kwargs_from_ini(ini_filepath_IF)
sim_prop_IF = SimulationProperties(**sim_kwargs_IF)
sim_prop_IF.load_steps(verbose=True)

star_1 = SingleStar(**{'mass':  16.563968, 'state': 'H-rich_Core_H_burning',
                'natal_kick_array': [250.739341, 0.378998, 1.529432, 1.176576]})
star_2 = SingleStar(**{'mass': 10.345701, 'state': 'H-rich_Core_H_burning',
                'natal_kick_array': [34.498945, 0.283107, 2.017168, 2.053916]})
binary_IF = BinaryStar(star_1, star_2, **{'time': 0.0,  'state': 'detached',  'event': 'ZAMS', 
                        'orbital_period': 5.995067,'eccentricity': 0.0}, properties = sim_prop_IF)
binary_IF.evolve()
```
This is running with IF and using IF to get the SN remnants.

**Comment by celiotine** [comment]
After discussing with @ZepeiX, it might be possible the new interpolators for the most recent grids were trained with an old pipeline script, I am looking into this now (should have been resolved w/ Issue #303).

Also, it is unclear to me if this issue is different than that in Issue #345. There are more of these systems in populations run with the 241011 grids compared to the 240920 grids, and BHs being mislabeled as NSs are only present in populations run with the 241011 grids.

**Comment by celiotine** [comment]
temporarily resolved with PR #416, but classification constraints still need to be added to the interpolator re discussion with @maxbriel @mkruckow @philipp-rajah
